src/Microsoft.Health.Fhir.SqlServer/Features/ChangeFeed/SqlServerFhirResourceChangeDataStore.cs

```diff
@@ -139,6 +139,11 @@ public async Task<IReadOnlyCollection<ResourceChangeData>> GetRecordsAsync(long
                     }
                 }
             }
+            catch (Exception ex) when ((ex is OperationCanceledException || ex is TaskCanceledException) && cancellationToken.IsCancellationRequested)
+            {
+                _logger.LogInformation(ex.Message);
+                return listResourceChangeData;
```

Comment:
If other prefer, we could instead opt to throw here without calling _logger.LogError and let the calling method deal with the exception.

In our use case, we are just going to call _logger.LogInformation anyway which is why I chose to call _logger.LogInformation here.

Reply:
updated to log an information and then throw an exception. added log message to string resx file